src/iframe-attributes.js

```diff
@@ -57,6 +60,14 @@ export function getContextMetadata(
       href: locationHref,
     },
     startTime,
+    tagName: element.tagName,
+    mode: getModeObject(),
+    canary: isCanary(parentWindow),
+    hidden: !viewer.isVisible(),
+    initialIntersection: element.getIntersectionChangeEntry ?
```

Comment:
all AMP created custom element should have this method: https://github.com/ampproject/amphtml/blob/master/src/custom-element.js#L1105